lib/workflows.js

```diff
@@ -5,15 +5,99 @@
  *
  */
 'use strict';
-
+const database = require('./database');
+const utils = require('./utils');
 const load = require('./workflows/load');
 const data = require('./workflows/data');
 
-module.exports = data;
 
-/**
- * Get all Workflows
+/*
+ * @typedef AuditEntry
+ * @type object
+ *
+ * @property {string} comment approver's comment
+ * @property {string} action  action taken.
+ * @property {number} step  current step in workflow
+ * @property {number} author  author of audit entry
+ * @property {FormattedDate} - The individually formatted date components
+ */
+
+/*
+ * An audit entry
+ *
+ * @param {string} comment approver's comment
+ * @param {string} action  action taken.
+ * @param {number} step  current step in workflow
+ * @param {object} req - HTTP Request
+ *
+ * @returns {AuditEntry} audit entry
+ */
+const entry = (comment, action, step, req) => {
+  const data = {
+    comment,
+    action,
+    step,
+    author: req.user.id,
+    created: utils.time.input(Date.now(), 'America/New_York'),
+  };
+
+  if ((typeof data.comment !== 'string') || (!data.comment)) {
+    data.comment = 'No comment';
+  }
+
+  if ((typeof data.action !== 'string') || (!data.action)) {
+    data.action = 'No action';
+  }
+
+  return data;
+}
+
+/*
+ * audits
  *
- * @returns {Promise} - An {Array} containing {Object} for each workflow, including `name` {String}, `id` name {String}, and steps {Object} loaded from YAML config file
+ * @params {object} revision
  */
-module.exports.raw = load;
+const audits = (revision, workflow, req) => {
+  const rev = revision;
+  const comment = req.body['comment--textarea'];
+  let audits = rev.audit;
+  let step;
+  let approval;
+  let publishable = false;
```

Comment:
Not a blocker, but I like to see all variables that have values assigned defined before those that are just initialted